pkg/controller/mysqlcluster/internal/cleaner/pvc.go

```diff
@@ -37,136 +34,124 @@ import (
 )
 
 const (
-	deleteSuccess        = "SucessfulDelete"
-	deleteFail           = "FailedDelete"
-	messagePvcDeleted    = "delete Claim %s in StatefulSet %s successful"
-	messagePvcNotDeleted = "delete Claim %s in StatefulSet %s failed"
+	reasonPVCClinupSuccess = "SucessfulPVCClinup"
+	reasonPVCClinupFail    = "FailedPVCClinup"
+	messagePVCDeleted      = "delete Claim %s in StatefulSet %s successful"
+	messagePVCNotDeleted   = "delete Claim %s in StatefulSet %s failed"
 )
 
 var log = logf.Log.WithName("mysqlcluster.pvccleaner")
 
-// PvcCleaner represents an object to clean Pvcs of a MysqlCluster
-type PvcCleaner struct {
-	cluster *mysqlcluster.MysqlCluster
-	opt     *options.Options
+// PVCCleaner represents an object to clean Pvcs of a MysqlCluster
+type PVCCleaner struct {
+	cluster  *mysqlcluster.MysqlCluster
+	opt      *options.Options
+	recorder record.EventRecorder
+	client   client.Client
 }
 
-// NewPvcCleaner returns a new PVC cleaner object
-func NewPvcCleaner(cluster *mysqlcluster.MysqlCluster, opt *options.Options) *PvcCleaner {
+// NewPVCCleaner returns a new PVC cleaner object
+func NewPVCCleaner(cluster *mysqlcluster.MysqlCluster, opt *options.Options, rec record.EventRecorder, c client.Client) *PVCCleaner {
 
-	return &PvcCleaner{
-		cluster: cluster,
-		opt:     opt,
+	return &PVCCleaner{
+		cluster:  cluster,
+		opt:      opt,
+		recorder: rec,
+		client:   c,
 	}
 }
 
 // Run performs cleanup of orphaned pvcs in a Mysql cluster
-func (p *PvcCleaner) Run(ctx context.Context, c client.Client, scheme *runtime.Scheme, recorder record.EventRecorder) error {
-	meta := p.cluster.ObjectMeta
-	key := types.NamespacedName{Name: meta.GetName(), Namespace: meta.GetNamespace()}
-
-	cluster := api.MysqlCluster{}
-	err := c.Get(ctx, key, &cluster)
-	if err != nil {
-		return err
-	}
+func (p *PVCCleaner) Run(ctx context.Context) error {
+	cluster := p.cluster.Unwrap()
 
 	if cluster.DeletionTimestamp != nil {
-		log.V(4).Info("being deleted, no action", "MysqlCluster", key)
+		log.V(4).Info("being deleted, no action", "MysqlCluster", p.cluster)
 		return nil
 	}
 
-	replicas := *cluster.Spec.Replicas
-
 	// Find any pvcs with higher ordinal than replicas and delete them
-	claims, err := p.getClaims(ctx, c)
+	claims, err := p.getClaims(ctx)
```

Comment:
`claims` -> `pvcs`. Claims is too general, as there can be many types of claims. We shoud use `PVC`, `pvc` so that you know that you are refering to a PersistentVolumeClaim.

So we should have `getPVCs`,  `deletePVC` and so on.